lib/matplotlib/backends/backend_webagg_core.py

```diff
@@ -236,8 +236,8 @@ def get_diff_image(self):
                 self._last_renderer, renderer)
             self._force_full = False
             self._png_is_old = False
-
-        return buff
+            self.buff = buff
+        return self.buff
```

Comment:
Maybe in `refresh_all` it should only send the diff if it is non None?  What does the consumer side do if it receives the same diff multiple times?

:-1: on looking at private attributes.